autorest/azure/azure.go

```diff
@@ -126,6 +131,58 @@ func ExtractRequestID(resp *http.Response) string {
 	return autorest.ExtractHeaderValue(HeaderRequestID, resp)
 }
 
+// GetAsyncOperation retrieves the long-running URL to poll from the passed response.
+func GetAsyncOperation(resp *http.Response) string {
+	return resp.Header.Get(http.CanonicalHeaderKey(HeaderAsyncOperation))
+}
+
+// ResponseIsLongRunning returns true if the passed response is for an Azure long-running operation.
+func ResponseIsLongRunning(resp *http.Response) bool {
+	return autorest.ResponseRequiresPolling(resp, http.StatusCreated) && len(GetAsyncOperation(resp)) > 0
```

Comment:
202 is handled by other code and, as per the current Azure REST specifications, relies on the Location header. If that's changed, then we can adapt this code.